### ISSUE TITLE
Allow PrivateUse1 backends to opt out of CompositeExplicitAutograd decomposition

### ISSUE THREAD

**Issue by vishalgoyal316** [pytorch/pytorch#187576]
## Problem

The PyTorch dispatch table resolves `CompositeExplicitAutograd` (CEA) kernels **before** the backend fallback.  PrivateUse1 backends that register a single catch-all fallback kernel never see composite ops — they only receive the already-decomposed primitives (e.g. `softmax` → `_softmax`, `convolution` → `convolution_overrideable`).

The resolution order today (`OperatorEntry::computeDispatchTableEntryWithDebug`):

```
1. Direct PrivateUse1 kernel           ← wins
2a. CEANF alias kernel                 ← applied before fallback
2b. CEA alias kernel                   ← applied before fallback
2c. CIA / Autograd alias kernels
3. Backend fallback                    ← only reached if no alias kernel
```

For **proxy backends** like [SkyTorch](https://github.com/astefanutti/skytorch) that forward ATen ops verbatim to a remote GPU server, receiving decomposed ops breaks the RPC protocol.  The server receives `_softmax` instead of `softmax`, requiring it to handle every decomposed primitive separately.

The only workaround today is to register every CEA-decomposed op explicitly at the `PrivateUse1` dispatch key.

## Scale

- **538** ATen op overloads have `CompositeExplicitAutograd` registrations in `native_functions.yaml`.
- SkyTorch currently registers **280+** ops manually to work around this.
- Each new PyTorch release can silently introduce new CEA decompositions that break the backend with no warning.

## Ask

Add a backend-level mechanism for PrivateUse1 backends to opt out of CEA decomposition, so that ops with only a CEA kernel route to the registered backend fallback instead of being silently decomposed.

## Proposed API

```python
# Python
handle = torch.utils.skip_cea_decomposition_for_privateuse1()
# Active while handle is alive; releasing it restores previous behavior.
del handle
```

Internally backed by `c10::Dispatcher::setPrivateUse1SkipCEA()` which:
1. Sets an atomic flag on the dispatcher.
2. Refreshes all `PrivateUse1` dispatch table entries (force-recomputes using the new resolution order where CEA alias lookup is skipped for `PrivateUse1`).
3. Returns an RAII handle for clean teardown.

With this flag set, the resolution order becomes:

```
1. Direct PrivateUse1 kernel           ← wins (unchanged)
2a. CEANF alias                        ← skipped for PrivateUse1
2b. CEA alias                          ← skipped for PrivateUse1
3. Backend fallback                    ← now reached for CEA-only ops
```

## Affected components

- `aten/src/ATen/core/dispatch/Dispatcher.h` / `.cpp` — new `setPrivateUse1SkipCEA()` API
- `aten/src/ATen/core/dispatch/OperatorEntry.cpp` — skip CEA/CEANF alias lookup when flag is set
- `torch/csrc/Module.cpp` — Python binding `torch._C._dispatch_set_privateuse1_skip_cea`
- `torch/utils/backend_registration.py` — `torch.utils.skip_cea_decomposition_for_privateuse1()`

**Comment by vishalgoyal316** [comment]
Closing - filed prematurely, not ready for review.
